homeassistant/components/telegram_bot/__init__.py

```diff
@@ -89,6 +307,255 @@ def async_platform_discovered(platform, info):
     return True
 
 
+class TelegramNotificationService(BaseNotificationService):
+    """Implement the notification services for the Telegram Bot domain."""
+
+    def __init__(self, hass, api_key, allowed_chat_ids, parser):
+        """Initialize the service."""
+        from telegram import Bot
+        from telegram.parsemode import ParseMode
+
+        self.allowed_chat_ids = allowed_chat_ids
+        self._default_user = self.allowed_chat_ids[0]
+        self._last_message_id = {user: None for user in self.allowed_chat_ids}
+        self._parsers = {PARSER_HTML: ParseMode.HTML,
+                         PARSER_MD: ParseMode.MARKDOWN}
+        self._parse_mode = self._parsers.get(parser)
+        self.bot = Bot(token=api_key)
+        self.hass = hass
+
+    def async_send_file(self, is_photo=True, **kwargs):
```

Comment:
There is no need to create these wrappers since no one is extending this class. Just have the service handler call everything in the executor.

```python
yield from hass.async_add_job(partial(notify_service.send_file, True, **kwargs))
```